pkg/controller/installation/core_controller.go

```diff
@@ -1783,3 +1779,35 @@ func removeInstallationFinalizer(i *operator.Installation) {
 		i.SetFinalizers(stringsutil.RemoveStringInSlice(CalicoFinalizer, i.GetFinalizers()))
 	}
 }
+
+func (r *ReconcileInstallation) updateInstallationDegraded(reason string, err error, instance *operator.Installation, ctx context.Context, log logr.Logger) {
+	r.SetDegraded(reason, err, log)
+	conditions := []operator.TigeraStatusCondition{
+		{Type: operator.ComponentDegraded, Status: operator.ConditionTrue, Reason: reason, Message: err.Error()},
+		{Type: operator.ComponentAvailable, Status: operator.ConditionFalse},
+		{Type: operator.ComponentProgressing, Status: operator.ConditionFalse},
+	}
+
+	for _, condition := range conditions {
```

Comment:
if instead you iterate over instance.status.conditions, I think you do not need to keep track of the ones that are found.
After the for-loop you can then do instance.status.conditions = conditions and update it.